src/test/groovy/org/prebid/server/functional/tests/TargetingSpec.groovy

```diff
@@ -1121,6 +1126,204 @@ class TargetingSpec extends BaseSpec {
+
+        where:
+        priceGranularity << PriceGranularityType.values() - UNKNOWN
+    }
+
+    def "PBS amp should prioritize price granularity from original request over account config"() {
+        given: "Default AmpRequest"
+        def ampRequest = AmpRequest.defaultAmpRequest
+
+        and: "Default ampStoredRequest"
+        def ampStoredRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.targeting = new Targeting(priceGranularity: PriceGranularity.getDefault(priceGranularity))
+            setAccountId(ampRequest.account)
+        }
+
+        and: "Create and save stored request into DB"
+        def storedRequest = StoredRequest.getStoredRequest(ampRequest, ampStoredRequest)
+        storedRequestDao.save(storedRequest)
+
+        and: "Account in the DB"
+        def accountAuctionConfig = new AccountAuctionConfig(priceGranularity: PBSUtils.getRandomEnum(PriceGranularityType))
+        def accountConfig = new AccountConfig(status: ACTIVE, auction: accountAuctionConfig)
+        def account = new Account(uuid: ampRequest.account, config: accountConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAmpRequest(ampRequest)
+
+        then: "BidderRequest should include price granularity from bidRequest"
+        def bidderRequest = bidder.getBidderRequest(ampStoredRequest.id)
+        assert bidderRequest?.ext?.prebid?.targeting?.priceGranularity == ampStoredRequest?.ext?.prebid?.targeting?.priceGranularity
+
+        where:
+        priceGranularity << PriceGranularityType.values() - UNKNOWN
+    }
+
+    def "PBS should include price granularity from account config when original request doesn't contain it"() {
+        given: "Default basic BidRequest"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.targeting = Targeting.createWithAllValuesSetTo(false)
+        }
+
+        and: "Account in the DB"
+        def accountAuctionConfig = new AccountAuctionConfig(priceGranularity: priceGranularity)
+        def accountConfig = new AccountConfig(status: ACTIVE, auction: accountAuctionConfig)
+        def account = new Account(uuid: bidRequest.accountId, config: accountConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "BidderRequest should include price granularity from account config"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest?.ext?.prebid?.targeting?.priceGranularity == PriceGranularity.getDefault(priceGranularity)
+
+        where:
+        priceGranularity << PriceGranularityType.values() - UNKNOWN
+    }
+
+    def "PBS should include price granularity from account config with different name case when original request doesn't contain it"() {
```

Comment:
"PBS auction should include price granularity from account config with different name case when original request doesn't contain price granularity"